File: 2073-time-needed-to-buy-tickets/2073-time-needed-to-buy-tickets.py
class Solution:
    def timeRequiredToBuy(self, tickets: List[int], k: int) -> int:
        q = deque()
        for ticket in tickets : 
            q.append(ticket)

        n = len(tickets)
        ans = 0
        i = 0
        while tickets[k] :
            if tickets[i] <= 0:
                i = (i +1)%n
                continue
            ans +=1
            q.popleft()
            
            if tickets[i] > 1:
                q.append(tickets[i]-1)
            tickets[i] -=1
            i= (i + 1)%n
                
        return ans

            

File: 2073-time-needed-to-buy-tickets/2073-time-needed-to-buy-tickets.cpp
class Solution {
public:
    int timeRequiredToBuy(vector<int>& tickets, int k) {
        queue<pair<int,int>> qp;
        for(int i = 0 ; i < (int)tickets.size() ; i++){
            qp.push({tickets[i], i});
        }
        int ans = 0;
        while(!qp.empty()){
            pair<int,int> element = qp.front();
            qp.pop();
            element.first --;
            ans ++;
            if(element.first > 0)
                qp.push(element);
            if(element.second == k && element.first == 0){
                break;
            }
        }
        return ans;
    }
};